prometheus_query_to_csv/prom_query_to_csv.py

```diff
@@ -0,0 +1,168 @@
+            json_data = json_file.read()
+            return json_data
+
+    @staticmethod
+    def json_to_dict(data: str) -> Dict:
+        """
+        This method converts the read string into a Python dictionary.
+        :param data: The data read from the file
+        :return: Returns the data as a dictionary
+        """
+        data = data.replace("'", '"')
+        json = loads(data)
+        return json
+
+    def dict_to_csv(self, json: Dict):
+        """
+        This method writes the data into a CSV file formatted using f strings to only write the data we want.
+        :param json: The data from the file in a dictionary
+        """
+        data = json["data"]["result"]
+        if data[0]["metric"]["__name__"].startswith("openstack"):
+            self.dict_to_csv_openstack(data)
+        elif data[0]["metric"]["__name__"].startswith("node"):
+            self.dict_to_csv_node(data)
+        else:
+            raise Exception(
+                "Unsupported query type: openstack or node currently supported."
+            )
+
+    @staticmethod
+    def dict_to_csv_openstack(data: Dict):
+        """
+        This method supports "openstack" queries.
+        """
+        file_name = data[0]["metric"]["__name__"] + ".csv"
+        with open(file_name, "w", encoding="utf-8") as csv_file:
+            line = f'Date Time Hostname {data[0]["metric"]["__name__"]}\n'
+            csv_file.write(line)
+            for metric in data:
+                for i in range(len(metric["values"]) - 1):
+                    time = datetime.fromtimestamp(metric["values"][i][0])
+                    hostname = metric["metric"]["hostname"]
+                    value = metric["values"][i][1]
+                    line = f"{time} {hostname} {value}\n"
+                    csv_file.write(line)
+
+    @staticmethod
+    def dict_to_csv_node(data: Dict):
+        """
+        This method supports "node" queries.
+        """
+        file_name = data[0]["metric"]["__name__"] + ".csv"
+        with open(file_name, "w", encoding="utf-8") as csv_file:
+            line = f'Date Time Hostname {data[0]["metric"]["__name__"]}\n'
+            csv_file.write(line)
+            for metric in data:
+                for i in range(len(metric["values"]) - 1):
+                    time = datetime.fromtimestamp(metric["values"][i][0])
+                    hostname = gethostbyaddr(
+                        metric["metric"]["instance"].split(":")[0]
+                    )[0]
+                    value = metric["values"][i][1]
+                    line = f"{time} {hostname} {value}\n"
+                    csv_file.write(line)
+
+
+if __name__ == "__main__":
+    metrics_to_query = [
+        "openstack_nova_vcpus_used",
+        "openstack_nova_memory_used_bytes",
+        "node_hwmon_power_average_watt",
+    ]
+    endpoint = "http://172.16.102.82:9090/api/v1/query_range"
+    RawData(
+        metrics_to_query, "1710770960", "1710857376", endpoint
```

Comment:
Nit: could we put what these hardcoded values are e.g.

```
# Start and end time as posix seconds - this represents x date and y date
start_time = "1710770960"
end_time = "1710857376"
```

(That's just me guessing)